ocean_lib/ocean/ocean_assets.py

```diff
@@ -602,9 +583,11 @@ def pay_for_compute_service(
             self._start_or_reuse_order_based_on_initialize_response(
                 datasets[i],
                 item,
-                consume_market_order_fee_address,
-                datasets[i].consume_market_order_fee_token,
-                datasets[i].consume_market_order_fee_amount,
+                FeeTokenInfo(
+                    consume_market_order_fee_address,
+                    datasets[i].consume_market_order_fee_token,
+                    datasets[i].consume_market_order_fee_amount,
+                ),
```

Comment:
I see that `consume_market_order_fee_address` is an argument to this method (`pay_for_compute_service`). Yet `consume_market_order_fee_token` and `.._amount` are in each dataset `datasets[i]`. 

Could we have it where: don't have argument `consume_market_order_fee_address`, and instead each dataset holds a `FeeTokenInfo` class ?

Reply:
Note: Calina said: There's no way to build the arguments beforehand, we only know the fee address until we retrieve the rest from provider

So I have my answer here. Resolving this.